code/go/0chain.net/chaincore/chain/state/state_context.go

```diff
@@ -137,6 +142,10 @@ type StateContext struct {
 	getSignature                  func() encryption.SignatureScheme
 	eventDb                       *event.EventDb
 	mutex                         *sync.Mutex
+	faucetscConfig                *SCConfig
+	storagescConfig               *SCConfig
+	vestingscConfig               *SCConfig
+	zcnscConfig                   *SCConfig
```

Comment:
We should not couple the SCs configs and the state package. This would not scale, which means we need to change the StateContext each time a new config need to be added. 

We actually don't need to expose through StateContext, instead keep the configs in SCs packages themself. I have thought about the necessary to update configs on finalization, and I think we don't have to, as it may cause problem if we do that. So what we will need to do is update the config in time when the MPT config is updated in SC packages.